3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -72,42 +45,21 @@ function initiate() {
   info "Successfully initialized replica set"
 }
 
-# Adds a host to the replica set configuration. It is safe to call this function
-# if the host is already in the configuration.
+# Adds a host to the replica set configuration.
 #
 # Arguments:
 # - $1: host address[:port]
 #
 # Global variables:
-# - MAX_ATTEMPTS
-# - SLEEP_TIME
 # - MONGODB_REPLICA_NAME
 # - MONGODB_ADMIN_PASSWORD
-# - WAIT_PRIMARY_OR_SECONDARY_STATUS
 function add_member() {
   local host="$1"
   info "Adding ${host} to replica set ..."
 
-  local script
-  script="
-    for (var i = 0; i < ${MAX_ATTEMPTS}; i++) {
-      var ret = rs.add('${host}');
-      if (ret.ok) {
-        quit(0);
-      }
-      // ignore error if host is already in the configuration
-      if (ret.code == 103) {
-        quit(0);
-      }
-      sleep(${SLEEP_TIME});
-    }
-    printjson(ret);
-    quit(1);
-  "
-
   # TODO: replace this with a call to `replset_addr` from common.sh, once it returns host names.
   local endpoints
-  endpoints="$(find_endpoints | paste -s -d,)"
+  endpoints="$(find_endpoints | grep -v $MEMBER_HOST | paste -s -d,)"
```

Comment:
Why using only local mongod address?

Connecting to replset with only this address won't work - this member is not initialized and connection with only this address will fail.